### ISSUE TITLE
Column Collation not being set in db:schema:load

### ISSUE THREAD

**Issue by pfriedman** [rails-sqlserver/activerecord-sqlserver-adapter#848]
## Issue

When attempting to set the collation for a column[1] that differs from the collation for the schema, the column collation gets set correctly when running `rake db:migrate` but running `rake db:schema:load` the column collation reverts back to the default schema location.

[1] https://docs.microsoft.com/en-us/sql/relational-databases/collations/set-or-change-the-column-collation

## Expected behavior

With a migration file that attempts to set the collation for a particular column, e.g. 

```
  change_column :models, :column_name, 'VARCHAR(256) COLLATE SQL_Latin1_General_CP1_CS_AS'
```

and the default collation being different (SQL_Latin1_General_CP1_CI_AS) (CI not CS) 
we expect that the collation for the column is the one we set in the change_column method

```
> Model.columns.select { |a| a.name == "column_name" }
=> [#<ActiveRecord::ConnectionAdapters::SQLServerColumn:0x0000...
  @collation="SQL_Latin1_General_CP1_CS_AS",
  @name="column_name",
```

ALSO in the `db/schema.rb` file we see that the collation comes across as a constant where it really should be a quoted String

```
t.varchar "column_name", limit: 256, collation: SQL_Latin1_General_CP1_CS_AS
```

where we'd expect it to look more like 

```
t.varchar "column_name", limit: 256, collation: "SQL_Latin1_General_CP1_CS_AS"
```

## Actual behavior
When running `rake db:schema:load`, the collation is set to the default 

```
> Model.columns.select { |a| a.name == "column_name" }
=> [#<ActiveRecord::ConnectionAdapters::SQLServerColumn:0x0000...
  @collation="SQL_Latin1_General_CP1_CI_AS",
  @name="column_name",
```

## Current Patch 
In our project we have the following change in order to "fix" this issue 

```
require 'activerecord-sqlserver-adapter'

# Constant set to bypass UninitializedConstant error
SQL_Latin1_General_CP1_CS_AS = 'SQL_Latin1_General_CP1_CS_AS'

module ActiveRecord
  module ConnectionAdapters
    module SQLServer
      class SchemaCreation
        def add_column_options!(sql, options)
          sql << " DEFAULT #{quote_default_expression(options[:default], options[:column])}" if options_include_default?(options)
          if options[:null] == false
            sql << " NOT NULL"
          end
          if options[:collation].present?
            sql << " COLLATE #{options[:collation]}"
          end
          if options[:is_identity] == true
            sql << " IDENTITY(1,1)"
          end
          if options[:primary_key] == true
            sql << " PRIMARY KEY"
          end
          sql
        end
      end
    end
  end
```

Basically patching the existing `add_column_options!` method to add 

```
          if options[:collation].present?
            sql << " COLLATE #{options[:collation]}"
          end
```

and also creating the constant **SQL_Latin1_General_CP1_CS_AS** so that we bypass an UninitializedConstant error when running `rake db:schema:load` 

## Details

- **Rails version**: `5.2.3`
- **SQL Server adapter version**: `5.2.0`
- **TinyTDS version**: `2.1.2`
- **FreeTDS details**:

  ```
  run `tsql -C` and paste here the output.
  ```
```
➜ tsql -C
Compile-time settings (established with the "configure" script)
                            Version: freetds v1.1.24
             freetds.conf directory: /usr/local/etc
     MS db-lib source compatibility: no
        Sybase binary compatibility: yes
                      Thread safety: yes
                      iconv library: yes
                        TDS version: 7.3
                              iODBC: no
                           unixodbc: yes
              SSPI "trusted" logins: no
                           Kerberos: yes
                            OpenSSL: yes
                             GnuTLS: no
                               MARS: yes
```

**Comment by gisborne** [comment]
I can take a look at this.

**Comment by wpolicarpo** [comment]
Closing this one as it was fixed in #881.
